components/utils/metadata.js

```diff
@@ -0,0 +1,197 @@
+import { sanitizeUrl } from './url'
+
+const fetchMetadata = (path) => {
+  const reqUrl = new URL(`/api/metadata`, window.location.origin)
+
+  const params = new URLSearchParams()
+  params.append('path', path)
+  reqUrl.search = params.toString()
+
+  return fetch(reqUrl)
+}
+
+export const inspectDataset = async (url) => {
+  // fetch zmetadata to figure out compression and variables
+  const sanitized = sanitizeUrl(url)
+
+  let response
+  try {
+    response = await fetchMetadata(sanitized)
+  } catch (e) {
+    // Show generic error message when request fails before response can be inspected.
```

Comment:
curious to hear your thoughts on logging the original error? it might come in handy for debugging

```suggestion
    // Show generic error message when request fails before response can be inspected.
    console.error(`Error fetching metadata for ${sanitized}:`, e);
```